ui/raidboss/browser_tts_engine.ts

```diff
@@ -96,13 +111,17 @@ export default class BrowserTTSEngine {
     }
   }
 
-  playSpeechTTS(text) {
-    this.ttsItems[text] = new SpeechTTSItem(text, this.speechLang, this.speechVoice);
-    this.ttsItems[text].play();
+  playSpeechTTS(text: string): void {
+    if (this.speechLang && this.speechVoice) {
+      const ttsItem = new SpeechTTSItem(text, this.speechLang, this.speechVoice);
```

Comment:
Up above, speechLang and speechVoice are optional, but here they're required.  What happens if these are undefined?

Reply:
That's my fault for not updating the constructor; the constructor can be made to take lang and voice as optional parameters, and this check isn't necessary.